File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/product/dto/ProductDetailResponse.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.product.dto;

import com.multicampus.gamesungcoding.a11ymarketserver.common.exception.DataNotFoundException;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.Product;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.ProductAiSummary;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.ProductImages;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.ProductStatus;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.SellerGrades;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record ProductDetailResponse(UUID productId,
                                    String productName,
                                    UUID sellerId,
                                    String sellerName,
                                    SellerGrades sellerGrade,
                                    Boolean isA11yGuarantee,
                                    Integer productPrice,
                                    ProductStatus productStatus,
                                    String productDescription,
                                    Integer productStock,
                                    List<ProductImageResponse> productImages,
                                    UUID categoryId,
                                    String categoryName,
                                    String summaryText,
                                    String usageContext,
                                    LocalDateTime submitDate,
                                    String usageMethod) {

    public static ProductDetailResponse fromEntity(Product product,
                                                   List<ProductImages> images,
                                                   ProductAiSummary summary) {
        if (product == null) {
            throw new DataNotFoundException("Product cannot be null");
        }

        if (images == null || images.isEmpty()) {
            images = java.util.List.of();
        }

        if (summary == null) {
            summary = ProductAiSummary.builder().build();
        }

        var seller = product.getSeller();
        var category = product.getCategory();

        return new ProductDetailResponse(
                product.getProductId(),
                product.getProductName(),
                seller.getSellerId(),
                seller.getSellerName(),
                seller.getSellerGrade(),
                seller.getIsA11yGuarantee(),
                product.getProductPrice(),
                product.getProductStatus(),
                product.getProductDescription(),
                product.getProductStock(),
                images.stream()
                        .map(ProductImageResponse::fromEntity)
                        .toList(),
                category.getCategoryId(),
                category.getCategoryName(),
                summary.getSummaryText(),
                summary.getUsageContext(),
                product.getSubmitDate(),
                summary.getUsageMethod()
        );
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/product/dto/ProductDetailResponse.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.product.dto

import com.multicampus.gamesungcoding.a11ymarketserver.feature.product.entity.ProductStatus
import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.SellerGrades
import java.time.LocalDateTime
import java.util.*

data class ProductDetailResponse(
    val productId: UUID,
    val productName: String,
    val sellerId: UUID,
    val sellerName: String,
    val sellerGrade: SellerGrades,
    val isA11yGuarantee: Boolean,
    val productPrice: Int,
    val productStatus: ProductStatus,
    val productDescription: String,
    val productStock: Int,
    val productImages: List<ProductImageResponse>,
    val categoryId: UUID,
    val categoryName: String,
    val summaryText: String? = null,
    val usageContext: String? = null,
    val submitDate: LocalDateTime,
    val usageMethod: String? = null
)
